translate/Main.hs

```diff
@@ -114,6 +114,36 @@ doTranslate signs circuit = do
             when (undef  /= []) $ putStr $ "The following signals have undefined initial states: \n"
                                     ++ unlines (map show undef) ++ "\n"
 
+handleArcs :: [([Transition DynSignal], Transition DynSignal)] -> [String]
```

Comment:
As a first version this is fine -- well done! But there are a few improvements that will simplify the code even further. Let's add a todo: `TODO: 1) Get rid of explicit recursion (use groupBy)? 2) Improve performance.`